pyEpiabm/pyEpiabm/sweep/demographics_sweep.py

```diff
@@ -0,0 +1,99 @@
+#
+# Sweep to record demographic information of the population in a .csv file
+#
+
+import os
+import typing
+
+from pyEpiabm.output import _CsvDictWriter
+
+from .abstract_sweep import AbstractSweep
+
+
+class DemographicsSweep(AbstractSweep):
+    """Class to sweep through the population at the beginning of the simulation
+    and record their demographic information
+
+    """
+
+    def __init__(self, file_params: typing.Dict):
+        """Initiate file parameters referring to the file location and
+        which columns are to be included
+
+        file_params Contains:
+            * `output_dir`: String for the location of the output file, \
+               as a relative path
+            * `spatial_output`: Boolean to determine whether a spatial output \
+               should be used
+            * `age_output`: Boolean to determine whether the output will \
+                have ages
+
+        """
+        if "output_dir" not in file_params:
+            raise ValueError("output_dir must be specified in file_params")
+        self.spatial_output = file_params["spatial_output"] \
+            if "spatial_output" in file_params else False
+        self.age_output = file_params["age_output"] \
+            if "age_output" in file_params else False
+
+        # Here we set up the writer
+        folder = os.path.join(os.getcwd(),
+                              file_params["output_dir"])
+        file_name = "demographics.csv"
+        self.titles = ["id"]
+        if self.age_output:
+            self.titles.append("age_group")
+        if self.spatial_output:
+            self.titles += ["location_x", "location_y"]
+        self.titles.append("kw_or_chr")
```

Comment:
Some of these titles aren't immediately obvious - is this for key workers and care home residents? Do document this somewhere if so

Reply:
There are different places you could do this, but in the docs for the class itself could be a good place. And then most importantly, have a working example so people know how to use this, in the python examples folder

Reply:
Yes, I will document this, and we are currently working on adapting the Gibraltar example to our code to hopefully see it working with different demographics, but will let you know how that goes!